packages/core/src/particle/ParticleGenerator.ts

```diff
@@ -503,6 +524,149 @@ export class ParticleGenerator {
+   */
+  _updateBoundsSimulationLocal(bounds: BoundingBox): void {
+    const { _localBounds: localBounds } = this._renderer;
+    if (this._renderer._isContainDirtyFlag(ParticleUpdateFlags.GeneratorVolume)) {
+      if (this.isAlive) {
+        this._calculateLocalBounds(localBounds);
+      } else {
+        localBounds.min.set(0, 0, 0);
+        localBounds.max.set(0, 0, 0);
+      }
+      this._renderer._setDirtyFlagFalse(ParticleUpdateFlags.GeneratorVolume);
+    }
+
+    if (this._renderer._isContainDirtyFlag(ParticleUpdateFlags.TransformVolume)) {
+      const { min: localMin, max: localMax } = localBounds;
+      this._addRotationAndVelocityOverLifetimeToBounds(localBounds);
+
+      const { min: worldMin, max: worldMax } = bounds;
+      const { _tempVector30: worldPosition } = ParticleGenerator;
+      worldPosition.copyFrom(this._renderer.entity.transform.worldPosition);
+
+      worldMin.set(localMin.x + worldPosition.x, localMin.y + worldPosition.y, localMin.z + worldPosition.z);
+      worldMax.set(localMax.x + worldPosition.x, localMax.y + worldPosition.y, localMax.z + worldPosition.z);
+
+      this._renderer._setDirtyFlagFalse(ParticleUpdateFlags.TransformVolume);
+    }
+
+    if (this._renderer._isContainDirtyFlag(RendererUpdateFlags.WorldVolume)) {
+      this._addGravityToBounds(bounds);
+      this._renderer._setDirtyFlagFalse(RendererUpdateFlags.WorldVolume);
+    }
+  }
+
+  /**
+   * @internal
+   */
+  _updateBoundsSimulationWorld(bounds: BoundingBox): void {
+    const { min, max } = bounds;
+    if (this._firstActiveBoundingBox === this._firstFreeBoundingBox) {
+      min.set(0, 0, 0);
+      max.set(0, 0, 0);
+      return;
+    }
+
+    if (this._renderer._isContainDirtyFlag(RendererUpdateFlags.WorldVolume)) {
+      const index = this._firstActiveBoundingBox * this._renderer.engine._particleBufferUtils.boundsFloatStride;
+
+      min.set(this._dynamicBounds[index], this._dynamicBounds[index + 1], this._dynamicBounds[index + 2]);
+      max.set(this._dynamicBounds[index + 3], this._dynamicBounds[index + 4], this._dynamicBounds[index + 5]);
+
+      if (this._firstActiveBoundingBox < this._firstFreeBoundingBox) {
+        for (let i = this._firstActiveBoundingBox; i < this._firstFreeBoundingBox; i++) {
+          this._mergeToWorldBounds(i, bounds);
+        }
+      } else {
+        for (let i = this._firstActiveBoundingBox; i < this._dynamicBoundsCapacity; i++) {
+          this._mergeToWorldBounds(i, bounds);
+        }
+        if (this._firstFreeBoundingBox > 0) {
+          for (let i = 0; i < this._firstFreeBoundingBox; i++) {
+            this._mergeToWorldBounds(i, bounds);
+          }
+        }
+
+        this._addGravityToBounds(bounds);
+        this._renderer._setDirtyFlagFalse(RendererUpdateFlags.WorldVolume);
+      }
+    }
+  }
+
+  /** @internal */
+  _resizeBoundsArray(isIncrease: boolean, increaseCount: number = ParticleGenerator._particleIncreaseCount): void {
+    if (isIncrease) {
```

Comment:
`ParticleGenerator._particleIncreaseCount` is particle increase count, i think use in bounds array is not OK